### PR TITLE
New module - ec2_vpc_net_facts

### PR DESCRIPTION
**Pull request by wimnat**
Facts module for vpc_net

Useful for getting vpc-id based on a tag.

@defionscode

### CODE REVIEW

**Comment by gregdek** [comment]
ping @jsmartin @scicoin-project @tombamford @garethr @lorin @jarv @jsdalton @silviud @adq @zbal @zeekin @willthames @lwade @carsongee @defionscode
@tastychutney @bpennypacker @loia

Thanks for submitting this new module.

To help facilitate reviews, we’ve broadened the number of people who can approve modules for inclusion into the Extras repository. The list of official reviewers can be found here:

https://github.com/ansible/ansible-modules-extras/blob/devel/REVIEWERS.md 

Our new policy is that if a new module is reviewed and approved by at least two official module reviewers, and receives no -1 votes, the module will be approved for inclusion.  We will be asking the community of reviewers to take a look at these modules on a regular basis.

To ensure that your module has the best chance of being approved, please double-check that you adhere to the Ansible module guidelines: http://docs.ansible.com/developing_modules.html#module-checklist

**Comment by Java1Guy** [comment]
I had done a similar thing in https://github.com/ansible/ansible-modules-extras/pull/937  -- at some point we might merge the two...

**Comment by robynbergeron** [comment]
Hi @wimnat @Java1Guy -- 

Just want to double-check this PR. It looks like #937 was related to this, as @Java1Guy pointed out, but that PR is now closed because #783 was merged -- I assume this one needs to continue to stay open, correct? :)

I am assuming so -- and thus, since this is a new module, and we did add some new policies for new modules as you both know, I'll paste that info in a separate comment so that folks are aware that they can review this for passing guidelines and functionality. I know you've both seen it by now but just trying to be more standard about things so newcomers can catch on :)

@Java1Guy -- this may be your cue to review this new module for those very things :)

**Comment by robynbergeron** [comment]
Thanks for submitting this new module to Ansible Extras! This module is now in community review, a process that is open to all Ansible users. In order for this module to be approved, it must gain the following votes:

“works_for_me”: If you have tested the module thoroughly, including testing of all of the module’s options, and if the module works for you, please add “works_for_me” in the comments.

“passes_guidelines”: If you have gone through the module guidelines and the module meets all of the requirements, please add “passes_guidelines” in the comments. Guidelines are available here: http://docs.ansible.com/developing_modules.html#module-checklist

“needs_revision”: If the module fails to work for you, or if it doesn’t meet guidelines, please add “needs_revision” in the comments with details about what needs to be fixed.

When a module has both “works_for_me” and “passes_guidelines” tags, we will promote the module for inclusion in Ansible Extras.  At this point, you will be expected to maintain the module by fixing bugs and evaluating pull requests in a timely manner. 

Thanks again for submitting your Ansible module!

**Comment by wimnat** [comment]
thanks @robynbergeron.  You are correct that this one needs to stay open.  Hopefully someone can pick up for review soon.

File: cloud/amazon/ec2_vpc_net_facts.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    aliases: [ 'aws_region', 'ec2_region' ]
+
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Gather facts about all VPCs
+- ec2_vpc_net_facts:
```
Comment (juliedavila) [inline cloud/amazon/ec2_vpc_net_facts.py:40]: In my testing it appears `region` is a required field, or is it being assumed the users boto profile has a region specified?

File: cloud/amazon/ec2_vpc_net_facts.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+# Gather facts about a particular VPC using VPC ID
+- ec2_vpc_net_facts:
+    filters:
+      - vpc-id: vpc-00112233
```
Comment (juliedavila) [inline cloud/amazon/ec2_vpc_net_facts.py]: Filters are dicts, not lists. This works by removing the hyphen in front of `vpc-id: vpc-00112233`

File: cloud/amazon/ec2_vpc_net_facts.py (hunk 1 above)
Comment (wimnat) [inline cloud/amazon/ec2_vpc_net_facts.py:40]: Replaced region with new ec2 doc fragment. It's not required as, as you pointed out, boto profile or environment variable could be used.

File: cloud/amazon/ec2_vpc_net_facts.py (hunk 2 above)
Comment (wimnat) [inline cloud/amazon/ec2_vpc_net_facts.py]: Grrrr! I keep doing this on my facts modules! Fixed!

**Comment by robynbergeron** [comment]
Hi @wimnat & @defionscode -- Just checking in on this. I see lots of magical code things going on (YAY) -- just a friendly reminder to @defionscode or any other kind soul who comes along that we need a works_for_me vote and a passes_guidelines vote to get this included.

I'm assuming for now that @defionscode will take a look at your updates as soon as humanly possible and will let us know when it works_for_him or passes guidelines, but friendly reminders are good sometimes nonetheless. :D

Thanks to both of you!

**Comment by juliedavila** [comment]
shipit
+1
passes_guidelines
works_for_me

I think I hit them all ;)

**Comment by robynbergeron** [comment]
@defionscode I am going to figure out a way to have :cocktail: in here somehow, someday :)

Marking for inclusion -- hooray, new module! cc @ansible/core team. ;) 

Exciting boilerplate follows. Thanks to @wimnat for this new module and to @defionscode for the reviews :)

This module has received all of the necessary votes to be included in Ansible Extras. Congratulations! 

A reminder: by submitting this module, you agree to maintain the module by fixing bugs and evaluating pull requests in a timely manner. If you are unable to do so, we may ask other community members to maintain the module, or we may deprecate the module altogether. 

Thanks again for contributing this module to the Ansible community.

**Comment by windowsrefund** [comment]
looks great. Can someone merge this into devel?

**Pull request merged**